pyop2/compilation.py

```diff
@@ -206,10 +220,34 @@ def get_so(self, src, extension):
                     # Atomically ensure soname exists
                     os.rename(tmpname, soname)
             # Wait for compilation to complete
+            newcomm.barrier()
             self.comm.barrier()
             # Load resulting library
             return ctypes.CDLL(soname)
 
+    @collective
+    def single_out_a_rank(self, comm):
+        rank = MPI.Comm.Get_rank(comm)
+        debug('Using file-based mechanism on %d', rank)
+        fname =  os.path.join("/tmp", "fb.%d" % rank)
+
+        with open(fname, 'w+') as f:
+            f.write("This is a file.")
+
+        MPI.Comm.Barrier(comm)
+        tbb = []
+        for dl,sl,fl in os.walk("/tmp"):
+            for fi in fl:
+                if str(fi).split(".")[0] == "fb":
+                    tbb.append(str(fi).split(".")[1])
+
+        MPI.Comm.Barrier(comm)
```

Comment:
I think this barrier is unnecessary: you're not going to try and communicate out of band any more.